src/roller.js

```diff
@@ -203,10 +215,18 @@ class LMRTFYRoller extends Application {
     }
     _onInitiative(event) {
         event.preventDefault();
+        this.message = "Rolling for Initiative"
         this._makeDiceRoll(event, game.system.data.initiative);
     }
     _onDeathSave(event) {
         event.preventDefault();
+        this.message = "Death Save"
+        this._makeDiceRoll(event, "1d20");
+    }
+
+    _onPerception(event) {
+        event.preventDefault();
+        this.message = "Perception Check"
```

Comment:
Don't override the `message`. That's a user provided message that the DM is sending to their players, so we should never override it.
You're also using it later to set the formula. It makes a lot more sense to set the formula here instead. See how initiative does it.
Use this instead of setting it to 1d20 then overriding the formula if the message matches the string...
```js
this._makeDiceRoll(event, `1d20 + @attributes.perception.totalModifier`);
```

Reply:
Well that was being used later but not the intent, the intent was to provide roll context, right now if you requested two rolls with MakeRoll they have no context. 

For example:
<img width="287" alt="Screenshot 2020-07-23 at 01 33 02" src="https://user-images.githubusercontent.com/14851464/88239786-a473f180-cc85-11ea-83c2-3493d137c164.png">

"Rolling for Initiative"

Otherwise it's just Amiri (Level 5) and a roll formula. I was trying to get the message into all three (Perception, Death Save, Initiative) for that context. Is this not the right place?

Reply:
Moved the block and used the @attributes.perception.totalModifier - my fault for not knowing the syntax of those. So that part is done, just have the query on usage of the .message now as I had put it in all three of the custom _makeDiceRoll chat message for context.

Reply:
Removed message overrides, I get what you are saying. I think a feature request I will open after this will be to add that flavor text if the message is already blank.

Reply:
Removed these overrides, will open a feature-request and maybe a p/r if accepted to add these messages in if the DM message is blank.

Reply:
Yeah, that makes sense, I'd say adding a `defaultMessage` as a 3rd argument to `makeDiceRoll` would make sense, then it could display `this.message || defaultMessage` and that's it. Would be nice indeed.